lib/tool-bar-plugin-manager.js

```diff
@@ -0,0 +1,61 @@
+'use babel';
+
+import ToolBarManager from './tool-bar-manager';
+import { CompositeDisposable } from 'atom';
+
+export default class ToolBarPluginManager {
+  plugins = {};
+  subscriptions = {};
+
+  constructor (config) {
+    this.config = config;
+    console.debug('ToolBarPluginManager::constructor', this.config);
+  }
+
+  register (name, toolBar) {
+    console.debug('ToolBarPluginManager::register', arguments);
+    let settingsKey = `tool-bar.plugins.${name}`;
+
+    const plugin = this.plugins[name] = new ToolBarManager(name, toolBar);
+    this.subscriptions[name] = new CompositeDisposable();
+
+    this.config.plugins[name] = {
+      type: 'boolean',
+      title: name,
+      description: `Whether the ${name} plugin is allowed to add items to the Tool Bar.`,
```

Comment:
Description isn't visible in the Settings anymore. Atom must have changed something internal. 
Possible solution: `atom.config.setSchema`.